closed/make/Javadoc.gmk

```diff
@@ -0,0 +1,475 @@
+JAVASE_BASE_URL := $(JDKJRE2COREAPI)
+
+################################################################################
+# Source files for openj9 extensions javadoc
+OPENJ9_DOCS_SRCDIR := $(JDK_OUTPUTDIR)/j9jcl_sources/jdk/src/share/classes
+
+# ALL_SOURCE_DIRS is used in the PackageFilter function
+ALL_SOURCE_DIRS += $(OPENJ9_DOCS_SRCDIR)
+
+#############################################################################
+# This file is included by (and is based on) make/Javadoc.gmk
+# Javadoc for the openj9 com.sun.management classes replaces the openjdk
+# classes documentation in doc/jre/api/management/extensions
+# Javadoc for the openj9 com.ibm.lang.management classes is created in
+# doc/jre/api/management/openj9extensions
+# Javadoc for other openj9 packages are created under docs/platform
+#############################################################################
+
+# html page footer for OpenJ9 extensions pages
+OPENJ9_BASE_URL := https://www.eclipse.org/openj9/
+OPENJ9_BUG_SUBMIT_URL := https://github.com/eclipse/openj9/issues
+OPENJ9_FULL_COMPANY_NAME := IBM Corp
+OPENJ9_JAVADOC_BOTTOM = \
+    <a href="$(OPENJ9_BASE_URL)">Link</a> to Eclipse OpenJ9 documentation.<br> \
+    To raise a bug report or suggest an improvement create an <a href="$(OPENJ9_BUG_SUBMIT_URL)">Eclipse Openj9 issue.</a><br> \
+    Copyright &copy; $(FIRST_COPYRIGHT_YEAR), $(COPYRIGHT_YEAR), $(OPENJ9_FULL_COMPANY_NAME). All rights reserved.
+
+#############################################################
+#
+# openj9-sharedclassesdocs
+#
+
+OPENJ9_SHAREDCLASSES_PKGDIR := $(OPENJ9_TOPDIR)/jcl/src/openj9.sharedclasses/share/classes
+OPENJ9_SHAREDCLASSES_PKGS := com.ibm.oti.shared
+
+OPENJ9_SHAREDCLASSES_FIRST_COPYRIGHT_YEAR := 1998
+
+OPENJ9_SHAREDCLASSES_DOCDIR := $(PLATFORM_DOCSDIR)/sharedclasses
+ifneq (,$(findstring openj9-docs-only,$(ALL_OTHER_TARGETS)))
+OPENJ9_SHAREDCLASSES2COREAPI := $(JAVASE_BASE_URL)
+else
+OPENJ9_SHAREDCLASSES2COREAPI := $(JDKJRE2COREAPI)
+endif
+OPENJ9_SHAREDCLASSES_DOCTITLE := OpenJ9 Shared Classes
+OPENJ9_SHAREDCLASSES_WINDOWTITLE := OpenJ9 Shared Classes
+OPENJ9_SHAREDCLASSES_HEADER := <strong>OpenJ9 Shared Classes</strong>
+#OPENJ9_SHAREDCLASSES_OVERVIEW := $(OPENJ9_SHAREDCLASSES_PKGDIR)/sharedclasses-overview.html  # Currently no overview.html for sharedclasses
+
+# The index.html, options, and packages files
+OPENJ9_SHAREDCLASSES_INDEX_FILE := $(OPENJ9_SHAREDCLASSES_DOCDIR)/index.html
+OPENJ9_SHAREDCLASSES_OPTIONS_FILE := $(DOCSTMPDIR)/openj9.sharedclasses.options
+OPENJ9_SHAREDCLASSES_PACKAGES_FILE := $(DOCSTMPDIR)/openj9.sharedclasses.packages
+
+openj9-sharedclassesdocs: $(OPENJ9_SHAREDCLASSES_INDEX_FILE)
+
+# Set relative location to platform document root
+$(OPENJ9_SHAREDCLASSES_INDEX_FILE): GET2DOCSDIR := $(PLATFORM_DOCSDIR)/..
+
+# Run javadoc if the index file is out of date or missing
+$(OPENJ9_SHAREDCLASSES_INDEX_FILE): $(OPENJ9_SHAREDCLASSES_OPTIONS_FILE) $(OPENJ9_SHAREDCLASSES_PACKAGES_FILE) coredocs
```

Comment:
Because `coredocs` is a phony target, this rule will alway run; likewise for other packages.

Reply:
Removed it as a dependency for the open9-xxx targets, but since openj9-mgmtdocs depends on the openjdk mgmtdocs which does have a dependency on coredocs, it will run anyway.